src/uwtools/drivers/driver.py

```diff
@@ -36,21 +37,25 @@ def __init__(
         self,
         cycle: Optional[datetime] = None,
         leadtime: Optional[timedelta] = None,
-        config: Optional[Union[dict, Path]] = None,
+        config: Optional[Union[dict, str, YAMLConfig, Path]] = None,
         dry_run: bool = False,
         key_path: Optional[list[str]] = None,
         schema_file: Optional[Path] = None,
+        coupler: Optional[str] = None,
     ) -> None:
-        self._config = YAMLConfig(config=config)
-        self._config.dereference(
+        self._config_full = config if isinstance(config, YAMLConfig) else YAMLConfig(config=config)
+        self._config_full.dereference(
             context={
-                **({"cycle": cycle} if cycle else {}),
-                **({"leadtime": leadtime} if leadtime is not None else {}),
-                **self._config.data,
+                **({STR.cycle: cycle} if cycle else {}),
+                **({STR.leadtime: leadtime} if leadtime is not None else {}),
+                **self._config_full.data,
             }
         )
+        self._config = self._config_full
         for key in key_path or []:
             self._config = self._config[key]
+        if coupler:
+            self._config[self._driver_name][STR.rundir] = self._config_full[coupler][STR.rundir]
```

Comment:
If a `coupler` name was provided, set the driver config's `rundir:` item from the named block.